### PR TITLE
Post refresh: Remove legacy color variables

### PR DESCRIPTION
**Pull request by umpox**
- Replace colors.scss with colors-redesign.scss
- Maps over usage of old `gray-N` values to specified alternatives from the [designs](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Design-Refresh-Systemization-source-of-truth?node-id=3222%3A44203)
- Replaces usage of colors that we no longer support. Suitable alternatives have been used (e.g. `icon-color` for icons, instead of non-semantic colors previously). There are a few areas where there is not a clear alternative, please see my comments below!
- Removes `web-content`, usage of this class will be removed in a follow-up PR

### CODE REVIEW

File: client/branded/src/global-styles/web-content.scss [hunk 1]
```diff
@@ -1,60 +0,0 @@
-// We intentionally want to style all descendants with these classes.
```
Comment (umpox) [inline client/branded/src/global-styles/web-content.scss:1]: I've deleted this file here as it was one of the main areas where we still relied on the SCSS variables. Further work to remove usage of `web-content` will be done in a different PR

File: client/shared/src/global-styles/icons.scss [hunk 2]
```diff
@@ -48,8 +48,8 @@ img.icon-inline,
 
 // @sourcegraph/react-loading-spinner coloring
 .theme-light {
-    --loading-spinner-outer-color: #{rgba($gray-19, 0.3)};
-    --loading-spinner-inner-color: #{rgba($gray-19, 1)};
+    --loading-spinner-outer-color: #{rgba($gray-08, 0.3)};
+    --loading-spinner-inner-color: #{rgba($gray-08, 1)};
```
Comment (umpox) [inline client/shared/src/global-styles/icons.scss:52]: Many changes here are following the redesign color mapping of gray values. e.g. see [here](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Design-Refresh-Systemization-source-of-truth?node-id=3222%3A44197). `gray-19` should map over to `gray-08`.

File: client/web/src/search/FeatureTour.module.scss [hunk 3]
```diff
@@ -1,10 +1,10 @@
 .feature-tour {
-    background: var(--tour-bg);
+    background: var(--color-bg-2);
     border: none;
     border-radius: 10px;
 
     :global(.shepherd-arrow::after) {
-        border-bottom-color: var(--tour-bg);
+        border-bottom-color: var(--color-bg-2);
```
[the remark this answers is not in the document]
Comment (limitedmage) [reply client/web/src/search/FeatureTour.module.scss:7]: `--tour-bg` needs to be added back as it does not match any of our existing background colors available in the redesign. It should be white in light mode. @sourcegraph/design can confirm the dark mode color.

File: client/branded/src/global-styles/colors.scss [hunk 4]
```diff
@@ -1,167 +1,207 @@
...
-    --secondary: #{$secondary-light};
-    --success: #{$success};
-    --info: #{$info};
-    --warning: #{$warning};
-    --danger: #{$danger};
-    --merged: #{$merged};
-    --color-border-active: #329af0;
-    // Logo colors
+    // Logo colors.
     --logo-orange: #f96216;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/colors.scss:78]: It seems that we don't use this color anywhere, but we have it on the Sticker Sheet.
Curios, do we expect to use it, @sourcegraph/design?

File: client/shared/src/global-styles/icons.scss (hunk 2 above)
Comment (valerybugakov) [reply client/shared/src/global-styles/icons.scss:52]: Is it possible to use CSS variables here instead of the `rgba` value? We discussed with the @sourcegraph/design that we want to limit colors to our color palette and avoid applying opacity values.

**Review by valerybugakov — APPROVED** [review]
Feels good!

File: client/branded/src/global-styles/colors.scss (hunk 4 above)
Comment (quinnkeast) [reply client/branded/src/global-styles/colors.scss:78]: Yep! It'll be used in the "progress indicators" in the post-signup flow:

CleanShot 2021-06-17 at 11 02 35@2x

Comment (valerybugakov) [reply client/branded/src/global-styles/colors.scss:78]: Got it, thanks @quinnkeast.

File: client/web/src/views/ChartViewContent/charts/line/constants.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 /**
  * Default value for line color in case if we didn't get color for line from content config.
  */
-export const DEFAULT_LINE_STROKE = 'var(--color-bg-5)'
+// TODO: Confirm usage, we want to deprecate this color
+export const DEFAULT_LINE_STROKE = 'var(--color-bg-3)'
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/views/ChartViewContent/charts/line/constants.ts:5]: @AlicjaSuska FYI

File: client/web/src/search/FeatureTour.module.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (umpox) [reply client/web/src/search/FeatureTour.module.scss]: Added `tour-bg` back in, used the figma file to map the old dark mode value to the new one
